src/test/mockFactory.ts

```diff
@@ -7,11 +7,11 @@ import PluginManager = require("serverless/classes/PluginManager");
 export class MockFactory {
   public static createTestServerless(config?: any): Serverless {
     const sls = new Serverless(config);
-    sls.service = MockFactory.createTestService();
-    sls.utils = MockFactory.createTestUtils();
-    sls.cli = MockFactory.createTestCli();
-    sls.pluginManager = MockFactory.createTestPluginManager();
-    sls.variables = {};
+    sls.service = config && config.service || MockFactory.createTestService();
```

Comment:
I think i know what you are trying to do here, but I don't believe it is working as you expect.  For example - if `config.service` is the defined then `sls.service` will be set to `true` not the value of `config.service`, otherwise it will use the `MockFactory.createTestService()`.

Reply:
Makes sense. I'll just add a ternary operator instead. Thanks